src/tracer.js

```diff
@@ -139,7 +139,11 @@ export default class Tracer {
 
     _report(span: Span): void {
         this._metrics.spansFinished.increment(1);
-        this._reporter.report(span);
+        try {
+          this._reporter.report(span);
+        } catch (err) {
+          this._logger.error(`could not report span ${span.operationName}: ${err.message}`)
+        }
```

Comment:
while it is tempting to protect all tracer methods with try-catch, doing so at the top level methods leads to Node engine marking them as [non-optimizable](https://community.risingstack.com/how-to-find-node-js-performance-optimization-killers/). The `report` method is called a lot more often than the actual place that was causing your exception, which was around the networking code, so I would rather move try-catch down the stack to the place that _actually can throw an exception_.